internal/storage/activities/delete_amss_test.go

```diff
@@ -0,0 +1,339 @@
+					handleRequest(w, r)
+				case "/api/v2/file/" + aipUUID + "/review_aip_deletion/":
+					w.Header().Set("Content-Type", "application/json")
+					_, err := w.Write([]byte("{"))
+					assert.NilError(t, err)
+				default:
+					t.Fatalf("unexpected request to %s", r.URL.Path)
+				}
+			},
+			wantErr: "approve deletion: failed to decode response body",
+		},
+		{
+			name:    "Fails approving deletion (response error_message)",
+			approve: true,
+			handler: func(w http.ResponseWriter, r *http.Request) {
+				switch r.URL.Path {
+				case "/api/v2/file/" + aipUUID + "/":
+					handleAIPInfo(w, r, "DEL_REQ")
+				case "/api/v2/file/" + aipUUID + "/delete_aip/":
+					handleRequest(w, r)
+				case "/api/v2/file/" + aipUUID + "/review_aip_deletion/":
+					w.Header().Set("Content-Type", "application/json")
+					err := json.NewEncoder(w).Encode(map[string]any{"error_message": "error message"})
+					assert.NilError(t, err)
+				default:
+					t.Fatalf("unexpected request to %s", r.URL.Path)
+				}
+			},
+			wantErr: "approve deletion: error message",
+		},
+		{
+			name: "Deletes AIP polling for status (approved)",
+			handler: func() http.HandlerFunc {
+				var requested, polled bool
+				return func(w http.ResponseWriter, r *http.Request) {
+					switch r.URL.Path {
+					case "/api/v2/file/" + aipUUID + "/":
+						if !requested {
+							handleAIPInfo(w, r, "DEL_REQ")
+							return
+						}
+						if !polled {
+							handleAIPInfo(w, r, "DEL_REQ")
+							polled = true
+							return
+						}
+						handleAIPInfo(w, r, "DELETED")
+					case "/api/v2/file/" + aipUUID + "/delete_aip/":
+						handleRequest(w, r)
+						requested = true
+					default:
+						t.Fatalf("unexpected request to %s", r.URL.Path)
+					}
+				}
+			}(),
+			want: activities.DeleteFromAMSSLocationActivityResult{Deleted: true},
+		},
+		{
+			name: "Deletes AIP polling for status (rejected)",
+			handler: func() http.HandlerFunc {
+				var requested, polled bool
+				return func(w http.ResponseWriter, r *http.Request) {
+					switch r.URL.Path {
+					case "/api/v2/file/" + aipUUID + "/":
+						if !requested {
+							handleAIPInfo(w, r, "DEL_REQ")
+							return
+						}
+						if !polled {
+							handleAIPInfo(w, r, "DEL_REQ")
+							polled = true
+							return
+						}
+						handleAIPInfo(w, r, "UPLOADED")
+					case "/api/v2/file/" + aipUUID + "/delete_aip/":
+						handleRequest(w, r)
+						requested = true
+					default:
+						t.Fatalf("unexpected request to %s", r.URL.Path)
+					}
+				}
+			}(),
+			want: activities.DeleteFromAMSSLocationActivityResult{Deleted: false},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			srv := setupServer(t, tt.handler)
+			ts := &temporalsdk_testsuite.WorkflowTestSuite{}
+			env := ts.NewTestActivityEnvironment()
+			env.SetTestTimeout(5 * time.Second)
+
+			// Set up a ticker channel for polling tests and tick twice.
+			ch := make(chan time.Time, 2)
```

Comment:
I think this kind of testing scenario where you are waiting for a timer to fire can now be handled without slowing down the tests or introducing an alternate timing channel by using https://go.dev/blog/synctest.  I haven't used synctest, but I think you could just wrap the test in a `synctest.Run()` function, then add `synctest.Wait()` (maybe twice?) after the `env.ExecuteActivity()` call.

Reply:
Oh, that's too bad.  Thanks for trying out the synctest package, hopefully it will come in handy in the future.